pkgs/applications/editors/vim/plugins/overrides.nix

```diff
@@ -301,6 +301,45 @@ self: super: {
     dependencies = with self; [ completion-nvim nvim-treesitter ];
   });
 
+  coq_nvim = super.coq_nvim.overrideAttrs (old: {
+    passthru.python3Dependencies = ps: [
+      ps.pynvim
+      ps.pyyaml
+      (ps.buildPythonPackage {
```

Comment:
In the main repo [there is a requirements.txt](https://github.com/ms-jpq/coq_nvim/blob/coq/requirements.txt) file. This requirements list which commit of those two packages it is using.

You should use that.

You can use a attrset which the keys are the rev commits and the values are the tarball hashes so if the requirements.txt changes in the future the build will break explicitly instead of just using the older version because the hash didn't change.

Reply:
> So I would undo those changes and go back to using the `ps.pynvim` and `ps.pyyaml` currently packaged in nixpkgs. Also it would be interesting to package `std2` as a proper derivation.

As you want. For `pyyaml`, I am already using the version of nixpkgs as is.
For the other two dependencies (`pynvim_pp` and `std2`), I could make separate commits to package them properly.
None of them are currently in nixpkgs.

Reply:
@lucasew what do you think ?

Reply:
> @lucasew what do you think ?

I'm with Thiago. I didn't realize in that time that we can't package inline python libraries.

Reply:
> Maybe, to avoid name conflicts we can add a prefix in the package name like coq_std2 but I am not sure if it's a good practice in this case.

Right now there are no packages named `std2` and `pynvim_pp` so we should maybe stick to the raw names...

Reply:
I see three reasons why not to package those dependencies:

1. Their use seem to be strictly limited to coq_nvim dependencies. They don't seem to have any value as is.
2. They are not packaged anywhere else (Pypi, distro...).
3. No description/license is provided on their repos.

Hence, for this specific case, we should maybe stick to the inline packaging method.

Reply:
> No description/license is provided on their repos.

If that is the case, we shouldn't even use them in nixpkgs. Or set the license to `unfree`, because this is basically what they are.

Reply:
> 



> If that is the case, we shouldn't even use them in nixpkgs. Or set the license to `unfree`, because this is basically what they are.

Or we can ask them to clarify the licensing issues.

Reply:
> Or we can ask them to clarify the licensing issues.

I did send a mail to the maintainer @ms-jpq.